src/lib.rs

```diff
@@ -162,3 +219,206 @@ impl LevelIndex {
+        SampleInformationIterator {
+            data: self.sample_data,
+            offset: 0,
+        }
+    }
+}
+
+struct SampleInformationIterator<'data> {
+    data: &'data [u8],
+    offset: usize,
+}
+
+impl<'data> Iterator for SampleInformationIterator<'data> {
+    type Item = SampleInformation;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.offset > self.data.len() - SampleInformation::LENGTH {
+            return None;
+        }
+        let sample_information =
+            SampleInformation::from_bytes(&self.data[self.offset..self.offset + SampleInformation::LENGTH]);
+        self.offset += SampleInformation::LENGTH;
+        Some(sample_information)
+    }
+}
+
+#[derive(Debug)]
+pub struct SampleInformation {
+    pub bit_offset: u32,                                //: 16;
+    pub bit_length: u32,                                //: 8;
+    pub channel_type: u32,                              //: 4;
+    pub channel_type_qualifiers: ChannelTypeQualifiers, //: 4;
+    pub sample_positions: [u32; 4],                     //: 8 x 4;
+    pub lower: u32,                                     //;
+    pub upper: u32,                                     //;
+}
+
+impl SampleInformation {
+    const LENGTH: usize = 16;
+
+    fn from_bytes(bytes: &[u8]) -> Self {
+        let mut offset = 0;
+
+        let v = bytes_to_u32(bytes, &mut offset);
+        let bit_offset = shift_and_mask_lower(0, 16, v);
+        let bit_length = shift_and_mask_lower(16, 8, v) + 1;
+        let channel_type = shift_and_mask_lower(24, 4, v);
+        let channel_type_qualifiers = ChannelTypeQualifiers::from_bits_truncate(shift_and_mask_lower(28, 4, v));
+
+        let v = bytes_to_u32(bytes, &mut offset);
+        let sample_positions = [
+            shift_and_mask_lower(0, 8, v),
+            shift_and_mask_lower(8, 8, v),
+            shift_and_mask_lower(16, 8, v),
+            shift_and_mask_lower(24, 8, v),
+        ];
+        let lower = bytes_to_u32(bytes, &mut offset);
+        let upper = bytes_to_u32(bytes, &mut offset);
+
+        assert_eq!(offset, Self::LENGTH);
+
+        Self {
+            bit_offset,
+            bit_length,
+            channel_type,
+            channel_type_qualifiers,
+            sample_positions,
+            lower,
+            upper,
+        }
+    }
+
+    pub fn is_exponent(&self) -> bool {
+        self.channel_type_qualifiers.contains(ChannelTypeQualifiers::EXPONENT)
+    }
+
+    pub fn is_float(&self) -> bool {
+        self.channel_type_qualifiers.contains(ChannelTypeQualifiers::FLOAT)
+    }
+
+    pub fn is_linear(&self) -> bool {
+        self.channel_type_qualifiers.contains(ChannelTypeQualifiers::LINEAR)
+    }
+
+    pub fn is_signed(&self) -> bool {
+        self.channel_type_qualifiers.contains(ChannelTypeQualifiers::SIGNED)
+    }
+
+    pub fn is_norm(&self) -> bool {
+        self.is_float() && self.upper == F32_1_AS_U32
+    }
```

Comment:
Do these offer significant value to callers, given that the fields and enums involved are already public and (presumably) correspond more directly to the spec?

Reply:
I'll just do it where I've been calling these. And so I'll remove the `F32_1_AS_U32` constant.